tests/phpunit/Application/QueryStringParserTest.php

```diff
@@ -53,33 +50,57 @@ public function testParsesNonExistenceConstraint(): void {
 		$this->assertTrue( $constraints->hasNoValue() );
 	}
 
-	public function testParsesItemTypes(): void {
-		$parser = $this->newQueryStringParser();
-		$query = $parser->parse( 'haswbstatement:' . self::INSTANCE_TYPE_ID . '=Q68' );
+	public function testParsesNonExistenceItemTypes(): void {
+		$parser = $this->newQueryStringParser( instanceOfId: 'P32' );
+		$query = $parser->parse( 'haswbfacet:P404' );
+
+		$this->assertEquals( [], $query->getItemTypes() );
+	}
+
+	public function testParsesSingleItemType(): void {
+		$parser = $this->newQueryStringParser( instanceOfId: 'P32' );
+		$query = $parser->parse( 'unrelated haswbstatement:P32=Q68 unrelated' );
 
 		$itemTypes = [
 			new ItemId( 'Q68' )
 		];
 
-		$this->assertEquals( $itemTypes, $query->getInstanceItemTypes() );
+		$this->assertEquals( $itemTypes, $query->getItemTypes() );
 	}
 
-	public function testParsesNonExistenceItemTypes(): void {
-		$parser = $this->newQueryStringParser();
-		$query = $parser->parse( 'haswbfacet:P42' );
+	public function testParsesMultipleItemTypes(): void {
+		$parser = $this->newQueryStringParser( instanceOfId: 'P32' );
+		$query = $parser->parse( 'unrelated haswbstatement:P32=Q68 haswbstatement:P32=Q67 unrelated haswbstatement:P32=Q69 unrelated' );
+
+		$itemTypes = [
+			new ItemId( 'Q68' ),
+			new ItemId( 'Q67' ),
+			new ItemId( 'Q69' ),
+		];
 
-		$this->assertEquals( [], $query->getInstanceItemTypes() );
+		$this->assertEquals( $itemTypes, $query->getItemTypes() );
 	}
 
 	public function testIgnoresHaswbstatementForNonInstanceOfIdProperties(): void {
-		$parser = $this->newQueryStringParser();
-		$query = $parser->parse( 'haswbstatement:P1=wrongId haswbstatement:' . self::INSTANCE_TYPE_ID . '=Q68 haswbstatement:P2=alsoWrong' );
+		$parser = $this->newQueryStringParser( instanceOfId: 'P32' );
+		$query = $parser->parse( 'haswbstatement:P1=wrongId haswbstatement:P32=Q68 haswbstatement:P2=alsoWrong' );
```

Comment:
I find this more readable than using a constant. By proving the ID in the line above, it is obvious what the expected ID is, and the query remains easy to read.

Reply:
That works too. On a side note, would it be more readable if the query string is broken down into an array and join together as a string?

Reply:
If the string are a bunch longer than perhaps a solution like that becomes worthwhile. Currently seems like a net negative